tcpproxy.go

```diff
@@ -33,6 +38,7 @@ func NewTCPProxy(c *gc.C, remoteAddr string) *TCPProxy {
 	p := &TCPProxy{
 		listener: listener,
 	}
+	p.stopStart = sync.NewCond(&p.mu)
```

Comment:
Or, if you declare stopStart as a value not a pointer, then:

    p.stopStart.L = &p.mu